File: Library/Types/BookModel.cs
namespace Library.Types
{
    public class BookModel
    {
        public BookModel(int id, string title, string author, Genre genre, int pages)
        {
            Id = id;
            Title = title;
            Author = author;
            Genre = genre;
            Pages = pages;
        }

        public int Id { get; set; }

        public string Title { get; set; } = null!;

        public string Author { get; set; } = null!;

        public Genre Genre { get; set; }

        public int Pages { get; set; }
    }
}

File: Library/ClientApp/src/app/models/BookModel.ts
export interface BookModel {
    id: number,
    title: String;
    author : String;
    genre : Genre; 
    pages : number;
}

export enum Genre {
    Fantasy = 1,
    Horror = 2,
    Science = 3,
    Sport = 4,
    Romance = 5
}